### PR TITLE
Get handling Functions

### PR DESCRIPTION
**Pull request by JLaRue96**
About 90% finished with this, just need to do some additional testing / bug fixing over the weekend.

### CODE REVIEW

File: syncr_tracker/tracker.py [hunk 1]
```diff
@@ -64,6 +63,89 @@ def handle_post(conn, request):
...
+
+def retrieve_drop_info(conn, drop_id):
+    """
+    Retrieves all current info related to a drop id.
+    :param conn: TCP socket connection between server and client.
+    :param drop_id: Dictionary key, which is also the drop id.
+    :return: list of node ip port tuples without timestamps to the client.
+    """
+    if drop_id not in drop_availability:
+        send_server_response(conn, ERROR_RESULT, 'Drop ID does not exist')
```
Comment (DLance96) [inline syncr_tracker/tracker.py]: Suggestion:
Changed error to Drop does not exist or is not currently available anywhere?

File: tests/test_get_requests.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import hashlib
+from unittest import mock
+from unittest.mock import MagicMock
+from unittest.mock import Mock
+
+from syncr_tracker.tracker import handle_get
+
+
```
Comment (DLance96) [inline tests/test_get_requests.py]: Add some additional tests

File: syncr_tracker/tracker.py [hunk 3]
```diff
@@ -64,6 +63,89 @@ def handle_post(conn, request):
...
+    """
+
+    if not os.path.exists(PUB_KEYS_DIRECTORY):
+        send_server_response(
+            conn, ERROR_RESULT,
+            'Public key does not exist for given Node ID',
+        )
+        pass
+
+    with open(generate_node_key_file_name(node_id), 'wb') \
```
Comment (matthewbentley) [inline syncr_tracker/tracker.py]: what if the file is not found?  The previous `if not...` statement only checks for the directory, not the key

**Comment by DLance96** [comment]
🦖

File: syncr_tracker/tracker.py [hunk 4]
```diff
@@ -64,6 +63,95 @@ def handle_post(conn, request):
...
+def retrieve_public_key(conn, node_id):
+    """
+    Retrieves the public key paired with the inputted node id.
+    :param conn: TCP socket connection between server and client
+    :param node_id: Node ID that is paired with a public key
+    :return: Message is sent to client with public key, if available
+    """
+
+    file_name = generate_node_key_file_name(node_id)
+    files = os.listdir(PUB_KEYS_DIRECTORY)
```
Comment (matthewbentley) [inline syncr_tracker/tracker.py]: this might/will fail if `os.pth.exists(PUB_KEYS_DIRECTORY)` is false

File: syncr_tracker/tracker.py [hunk 5]
```diff
@@ -64,6 +63,95 @@ def handle_post(conn, request):
...
+
+def retrieve_public_key(conn, node_id):
+    """
+    Retrieves the public key paired with the inputted node id.
+    :param conn: TCP socket connection between server and client
+    :param node_id: Node ID that is paired with a public key
+    :return: Message is sent to client with public key, if available
+    """
+
+    file_name = generate_node_key_file_name(node_id)
```
Comment (matthewbentley) [inline syncr_tracker/tracker.py]: The file name from `generate_node_key_file_name` will include the Directory

**Pull request merged**